packages/@ember/-internals/glimmer/tests/integration/helpers/gt-test.js

```diff
@@ -0,0 +1,103 @@
+import { RenderingTest, moduleFor } from '../../utils/test-case';
+import { set } from '@ember/-internals/metal';
+
+moduleFor(
+  'Helpers test: {{gt}}',
+  class extends RenderingTest {
+    ['@test returns true when the first static argument is bigger than the second one']() {
+      this.render(`{{gt 2 1}}`);
+      this.assertText('true');
+    }
+
+    ['@test returns false when the first static argument is not bigger than the second one']() {
+      this.render(`{{gt 0 1}}`);
+      this.assertText('false');
+    }
+
+    ['@test returns false when both arguments are the same']() {
+      this.render(`{{gt 1 1}}`);
+      this.assertText('false');
+    }
+
+    // ['@test it coerces the first argument from a string into a number']() {
+    //   this.render(`{{gt "2" 1}}`);
+    //   this.assertText('true');
+    // }
+
+    // ['@test it coerces the second argument from a string into a number']() {
+    //   this.render(`{{gt 2 "1"}}`);
+    //   this.assertText('true');
+    // }
+
+    ['@test it updates for bound arguments']() {
+      this.render(`{{gt left right}}`, { left: 1, right: 2 });
+
+      this.assertText('false');
+
+      this.runTask(() => this.rerender());
+
+      this.assertText('false');
+
+      this.runTask(() => set(this.context, 'left', 3));
+
+      this.assertText('true');
+
+      this.runTask(() => set(this.context, 'right', 4));
+
+      this.assertText('false');
+
+      this.runTask(() => set(this.context, 'left', 5));
+
+      this.assertText('true');
+
+      this.runTask(() => set(this.context, 'right', 5));
+
+      this.assertText('false');
+    }
+
+    ['@test it can be used as a sub-expression']() {
+      this.render(`{{if (gt left right) "yes" "no"}}`, { left: 1, right: 2 });
+
+      this.assertText('no');
+
+      this.runTask(() => this.rerender());
+
+      this.assertText('no');
+
+      this.runTask(() => set(this.context, 'left', 3));
+
+      this.assertText('yes');
+
+      this.runTask(() => set(this.context, 'right', 4));
+
+      this.assertText('no');
+
+      this.runTask(() => {
+        set(this.context, 'left', 5);
+        set(this.context, 'right', 0);
+      });
+
+      this.assertText('yes');
+    }
+
+    ['@test of the first argument is null or undefined, it never pulls the second argument']() {
+      let didInvokeRight = false;
+      this.render(`{{gt left right}}`, {
+        left: undefined,
+        get right() {
+          debugger;
```

Comment:
@rwjblue Any guidance on how to define a getter or a CP to verify that its value is not eagerly pulled by the helper? Seems that right now the keys of this object are eagerly accessed by `Object.assign` by the setup of the test itself.

Reply:
Instead of rendering gt directly render a component whose own template uses gt, then have that component have left and right CP’s that you can track when they are evaluated